### DISCUSSION TITLE
Cannot retrieve child headers

### DISCUSSION THREAD

**Discussion by TheBrambleShark** [statiqdev/Discussions#45] [Q&A] [answered]
I have created a file called `sample.md` to demonstrate this issue. Contents are as follows:
```md
---
Layout: _DocsLayout
---
# Title

Some text

## Section 1

Section 1 text

## Section 2

Section 2 text
```

In the _DocsLayout page, I specify a sidebar which holds this content:

```cshtml
@{
    var headings = Document.GetDocumentList(Statiq.Html.HtmlKeys.Headings);
}
@* snip *@
<div class="box">
    <ul>
        @foreach (IDocument heading in headings)
        {
            <li>@(await heading.GetContentStringAsync())</li>
            <ul>
                @{
                    var documentListCount = heading.GetDocumentList(Statiq.Html.HtmlKeys.Headings).Count;
                    var childCount = heading.GetChildren().Count;
                    <li>Document List Count: @documentListCount</li>
                    <li>Child Count: @childCount</li>
                }
            </ul>
        }
    </ul>
</div>
```
When I render this page, the box appears as this:
![Screenshot_2021-01-24 Sample -- LuzFaltex](https://user-images.githubusercontent.com/7003081/105640044-c38d4f00-5e73-11eb-96d3-5a3fba7d38a3.png)

It seems the engine correctly identifies `<h1>Title</h1>` as being a header element, but this is where it stops. The [headings document](https://statiq.dev/web/content-and-data/content/excerpts-and-headings#headings) on the website suggests that child headings are stored as children. Is the system not properly detecting H2 elements as being headers or am I misunderstanding the website as far as how to get child elements? I've been experimenting with this for a bit trying to find a way to get to them but with no luck.

Edit: I have manually pushed to GitHub Pages if you wish to see an example: https://luzfaltex.github.io/policies/terms
https://github.com/LuzFaltex/luzfaltex.github.io/blob/e2288dd06d71ec176570ffde7ced7660637acf5d/input/Shared/_tocList.cshtml#L11-L34

**Comment by TheBrambleShark** [comment, marked as answer]
Found the fix for this. In my settings, set `GatherHeadingsLevel` to how many layers deep you want. In my case, I chose 3. It now detects child headings properly on all Markdown pages. On HTML pages, it does not appear to detect any headings, so I may need to do some tinkering here.

**Reply by daveaglick** [reply]
Just commented in the other discussion that should help clarify some of this.

**Reply by TheBrambleShark** [reply]
Tagging #58 for others who find this later.
